src/OpenSpaceNet.cpp

```diff
@@ -614,26 +629,63 @@ void OpenSpaceNet::skipLine() const
     }
 }
 
-SizeSteps OpenSpaceNet::calcSizes() const
+cv::Size OpenSpaceNet::calcPrimaryWindowSize() const
 {
-    if(args_.pyramidWindowSizes.empty()) {
-        if(args_.pyramid) {
-            return SlidingWindow::calcSizes(bbox_.size(), model_->metadata().windowSize(), stepSize_, 2.0);
-        } else {
-            return { { model_->metadata().windowSize(), stepSize_ } };
-        }
-    } else {
-        // Sanity check, should've been caught before
-        DG_CHECK(args_.pyramidWindowSizes.size() == args_.pyramidStepSizes.size(), "Pyramid window sizes don't match step sizes.");
+    const auto& modelSize = model_->metadata().modelSize();
+    auto windowSize = model_->metadata().modelSize();
+    if(!args_.windowSizes.empty()) {
+        windowSize = {args_.windowSizes[0] * modelSize.height / modelSize.width, args_.windowSizes[0]};
```

Comment:
It may be useful to pre-calculate `(float)modelSize.height / modelSize.width` and store it as `heightRatio` or something, just to make the code a bit cleaner.